### PR TITLE
fix: uniqueness and non-emptiness validation of column or tag names

### PR DESCRIPTION
**Pull request by bgzhao66**
Cod change is the same as [PR-14182](https://github.com/taosdata/TDengine/pull/14182), but against v2.6 instead of v2.4.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,7 @@ cmake-build-debug/
 cmake-build-release/
 cscope.out
 cscope.files
+tags
```
Comment (bgzhao66) [inline .gitignore:9]: Ignore ctags index file `tags`

File: src/client/src/tscSQLParser.c [hunk 2]
```diff
@@ -1720,8 +1720,8 @@ static bool validateTagParams(SArray* pTagsList, SArray* pFieldList, SSqlCmd* pC
   // field name must be unique
   for (int32_t i = 0; i < numOfTags; ++i) {
     TAOS_FIELD* p = taosArrayGet(pTagsList, i);
-
-    if (has(pFieldList, 0, p->name) == true) {
+    size_t      numOfCols = taosArrayGetSize(pFieldList);
+    if (has(pFieldList, numOfCols, p->name) == true) {
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:1724]: check the slice [0, numOfCols) of pFieldList for existence of p->name

File: src/client/src/tscSQLParser.c [hunk 3]
```diff
@@ -1867,9 +1867,8 @@ int32_t validateOneColumn(SSqlCmd* pCmd, TAOS_FIELD* pColField) {
 }
 
 /* is contained in pFieldList or not */
-static bool has(SArray* pFieldList, int32_t startIdx, const char* name) {
-  size_t numOfCols = taosArrayGetSize(pFieldList);
-  for (int32_t j = startIdx; j < numOfCols; ++j) {
+static bool has(SArray* pFieldList, int32_t endIdx, const char* name) {
+  for (int32_t j = 0; j < endIdx; ++j) {
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:1871]: Check the slice [0, endIdx) of pFieldList which has been processed, for existence of name,

File: src/client/src/tscSQLParser.c [hunk 4]
```diff
@@ -8012,7 +8011,9 @@ int32_t validateColumnName(char* name) {
     return validateColumnName(token.z);
   } else if (token.type == TK_ID) {
     stringProcess(name, token.n);
-    return TSDB_CODE_SUCCESS;
+    if (strlen(name) == 0) {
+      return TSDB_CODE_TSC_INVALID_OPERATION;
+    }
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:8016]: Check non-emptiness of name after stringProcess.

File: src/client/src/tscSQLParser.c [hunk 5]
```diff
@@ -1632,7 +1632,7 @@ static bool validateTableColumnInfo(SArray* pFieldList, SSqlCmd* pCmd) {
     }
 
     // field name must be unique
-    if (has(pFieldList, i + 1, pField->name) == true) {
+    if (has(pFieldList, i, pField->name) == true) {
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:1635]: Check the slice [0, i) of pFieldList.

File: src/client/src/tscSQLParser.c [hunk 6]
```diff
@@ -1691,7 +1691,7 @@ static bool validateTagParams(SArray* pTagsList, SArray* pFieldList, SSqlCmd* pC
       return false;
     }
 
-    if (has(pTagsList, i + 1, p->name) == true) {
+    if (has(pTagsList, i, p->name) == true) {
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:1694]: Check the slice [0, i) of pTagsList

**Review by DuanKuanJun — APPROVED** [review]

**Pull request merged**